src/printer.ml

```diff
@@ -0,0 +1,52 @@
+open View
+open Operator
+open Abt_inner
+
+module type ABT_PRINTER = sig
+  module Abt : ABT
+  module View : VIEW
+
+  val print : Abt.t -> string
+end
+
+module type OPERATOR_PRINTER = sig
+  type op
+
+  val print : op -> string
+end
+
+module MakeSexprPrinter
+  (O : OPERATOR_PRINTER)
+  (V : VIEW with type op = O.op)
+  (A : ABT with type View.op = O.op and type View.Variable.t = V.Variable.t)
+  (W : Prettiest.Width)
+  : ABT_PRINTER
+  with type View.op = V.op
+  and type Abt.t = A.t
+  = struct
+
+  module Option = Base.Option
+
+  module View = V
+  module Abt = A
+  module P = Prettiest.Make(W)
+
+  open View
+  open P.Infix
+
+  type t = A.t
+
+  let rec toPretty e =
+    match A.out e with
+    | VarView x -> P.text (Variable.toUserString x)
+    | AbsView (x, e) -> P.text (Variable.toUserString x) <> P.text " . " <> toPretty e
```

Comment:
I prefer no space in front the dot, but that's a matter of taste I guess :-)
Prettiest.Characters defines a dot you can use if you like.

Reply:
Yeah, I think that for the default printer it's better to have that space; dot surrounded by spaces is easier to spot.